WordPress/Classes/ViewRelated/System/WPTabBarController.m

```diff
@@ -278,6 +279,26 @@ - (void)showPostTabWithOptions:(NSDictionary *)options
     [self presentViewController:navController animated:YES completion:nil];
 }
 
+- (void)showLoginController
+{
+    LoginViewController *loginViewController = [[LoginViewController alloc] init];
+    loginViewController.showEditorAfterAddingSites = YES;
+    loginViewController.cancellable = YES;
+    
+    NSManagedObjectContext *context = [[ContextManager sharedInstance] mainContext];
+    AccountService *accountService = [[AccountService alloc] initWithManagedObjectContext:context];
+    WPAccount *defaultAccount = [accountService defaultWordPressComAccount];
+    
+    if (!defaultAccount) {
+        loginViewController.prefersSelfHosted = YES;
+    }
+    loginViewController.dismissBlock = ^{
+        [self dismissViewControllerAnimated:YES completion:nil];
+    };
+    UINavigationController *loginNavigationController = [[UINavigationController alloc] initWithRootViewController:loginViewController];
+    [self presentViewController:loginNavigationController animated:YES completion:nil];
+}
+
```

Comment:
@knutigro hello there!,

I've noticed that this new `showLoginController` method is roughly the same as the old `showWelcomeScreenAnimated:thenEditor:` selector.

Since the only difference between them is the `cancellable` flag, perhaps it would make sense to, instead, [update the logic here](https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/System/WordPressAppDelegate.m#L527), so that the view is cancellable if the user has a dotcom account with no blogs.

Thanks for helping us out!